trace.py

```diff
@@ -37,26 +37,40 @@ def _get_packets(self, f: str) -> List:
     return packets
 
   def get_1rtt(self, direction: Direction = Direction.ALL) -> List:
-    """ Get all QUIC packets, one or both directions.
-    """
-    return self._get_packets(self._get_direction_filter(direction) + "quic.header_form==0")
-
+    """ Get all QUIC packets, one or both directions. """
+    packets = []
+    for packet in self._get_packets(self._get_direction_filter(direction) + "quic.header_form==0"):
+      for layer in packet.layers:
+        if layer.layer_name == "quic" and not hasattr(layer, "long_packet_type"):
+          layer.sniff_time = packet.sniff_time
```

Comment:
This might be considered a bit hacky 🙃

Reply:
No, it's uglier than this. It's adding a field that didn't exist before.

Before, this function used to return a `[ IP + UDP + QUIC ]`, or in the case of a coalesced packet a `[ IP + UDP + { QUIC + QUIC } ]` packet.
To properly deal with coalesced packets, we now split up the layers of the packet and only return `[ QUIC ]`. However, `sniff_time` is only set on the IP packet (iirc), not on the higher layers.